crates/wasmi/src/engine/executor.rs

```diff
@@ -92,15 +93,25 @@ pub enum ReturnOutcome {
 ///
 /// If the Wasm execution traps.
 #[inline(never)]
-pub fn execute_wasm<'engine>(
-    ctx: &mut StoreInner,
+pub fn execute_wasm<'ctx, 'engine>(
+    ctx: &'ctx mut StoreInner,
     cache: &'engine mut InstanceCache,
     value_stack: &'engine mut ValueStack,
     call_stack: &'engine mut CallStack,
     code_map: &'engine CodeMap,
     const_pool: ConstPoolView<'engine>,
+    resource_limiter: ResourceLimiterRef<'ctx>,
```

Comment:
For the sake of experimentation with benchmarks we could use `&mut ResourceLimiterRef<'ctx>` here instead since that's how it is used anyways.

Reply:
Will try.

Reply:
Yes, this works. Pushed a version that does so. (Maybe that'll help with the benchmarks? Fingers crossed)

Reply:
Seems like it didn't help. Another thing I often succesfully try is to toy around with different kinds of `#[inline]`, `#[inline(always)]` and `#[inline(never)]` to guide the codegen towards hot patchs and cold paths. It is super ugly but sometimes it gets the job done. A perfect display of how `match+loop` isn't a good fit for efficient interpreters.